osi_common.proto

```diff
@@ -428,3 +428,210 @@ message BaseMoving
     //
     repeated Vector2d base_polygon = 7;
 }
+
+//
+// \brief A description for the steeringwheel.
+//
+message Steeringwheel
+{
+    // Angle of the steeringwheel. 
+    //
+    // Unit in rad: 0=Central (Straight); Left>0; 0>Right.
+    //
+    optional double angle = 1;
+
+    // Angle-speed of the steeringwheel.
+    //
+    // Unit in rad/s: 0=Central (Straight); Left>0; 0>Right.
+    //
+    optional double angular_speed = 2;
+
+    // Torque of the steeringwheel to the hand.
+    //
+    // Unit in Nm: 0=Central (Straight); Left>0; 0>Right.
+    //
+    optional double torque = 3;
+}
+
+//
+// \brief A description for the positions of the pedals.
+//
+//
+message Pedalry
+{
+    // Position of the acceleration-pedal.
+    // 0-100 (percentage of position: Unpressed - fully pressed)
+    //
+    optional double pedal_position_acceleration = 1;
+
+    // Position of the brake-pedal.
+    // 0-100 (percentage of position: Unpressed - fully pressed)
+    //
+    optional double pedal_position_brake = 2;
+
+    // Position of the clutch-pedal.
+    // 0-100 (percentage of position: Unpressed - fully pressed)
+    //
+    optional double pedal_position_clutch = 3;
+}
+
+//
+// \brief This is a message to describe, which trajectory the vehicle should follow.
+//
+//
+message Trajectory
+{
+    // Contains the timestamp where the trajectorypoint should be reached.
+    // In [s].
+    //
+    optional Timestamp timestamp = 1;
+
+    // Contains the X-Position the vehicle should be at the timestamp.
+    //
+    optional double targeted_pos_x = 2;
+
+    // Contains the Y-Position the vehicle should be at the timestamp.
+    //
+    optional double targeted_pos_y = 3;
+
+    // Direction of the vehicle at the timestamp.
+    // In [Rad].
+    //
+    optional double trackangle = 4;
+
+    // Contains the curvature at the timestamp.
+    // In [1/m].
+    //
+    optional double curvature = 5;
+
+    // Contains the curvaturechange at the timestamp.
+    // In [1/m*s].
+    //
+    optional double curvaturechange = 6;
+
+    // Contains the velocity of the vehicle at the timestamp.
+    // In [m/s].
+    //
+    optional double velocity = 7;
+
+    // Contains the acceleration of the vehicle at the timestamp.
+    // In [m/s^2].
+    //
+    optional double acceleration = 8;
+
+    // Contains the interpolation method.
+    //
+    optional InterpolationMethod interpolation_method = 9;
+
+    // Contains the interpolation method.
+    //
+    enum InterpolationMethod
+    {
+        // Stay on the actual lane.
+        //
+        INTERPOLATION_METHOD_LINEAR = 0;
+
+        // Change to the left.
+        //
+        INTERPOLATION_METHOD_CUBIC = 1;
+    }
+}
+
+// The actual gear of the car.
+//
+enum Gear
```

Comment:
ppannusch on 11 Dec 2018  Member
What about M/S? Or is this what you mean by GEAR_MID?
Also the enum mixes up actual gear of the powertrain and the gear lever state. What about shift paddles?

Reply:
==> Open Point